src/task_queue/database_tasks/contents_spacing_correction.py

```diff
@@ -0,0 +1,110 @@
+import csv
+import datetime
+from django.apps import apps
+from sentry_sdk import capture_message
+from _main_.utils.emailer.send_email import send_massenergize_email_with_attachments
+from api.utils.constants import DATA_DOWNLOAD_TEMPLATE
+from database import models as db_models
+from django.http import HttpResponse
+import re
+
+
+FEATURE_FLAG_KEY = "update-html-format-feature-flag"
+PATTERNS = ["<p><br></p>", "<p>.</p>", "<p>&nbsp;</p>"]
+PATTERN = "|".join(re.escape(p) for p in PATTERNS)
+
+"""
+This File is used to fix spacing for both universal contents(contents not tied to a community) and content tied to a community.
+There are two parts to this:
+1. Generate a report of the contents that need to be fixed with the following information
+    a. Community: the community the content belongs to
+    b. Content type: whether content is an Action/Event/Testimonial etc.
+    c. Item name: the name or title of the content
+    d. Field name: the field name of the content to be corrected
+    e. Count: the number of occurrences of spacing in the content.
+2. Fix the spacing in the database.
+"""
+
+def write_to_csv(data):
+    response = HttpResponse(content_type="text/csv")
+    writer = csv.DictWriter(response, fieldnames=["Community", "Content Type", "Item Name", "Field Name", "Count"])
+    writer.writeheader()
+    for row in data:
+        writer.writerow(row)
+    return response.content
+def get_community(instance):
+    if instance and hasattr(instance, "community"):
+        return instance.community.name if instance.community else ""
+    elif instance and hasattr(instance, "primary_community"):
+        return instance.primary_community.name if instance.primary_community else ""
+    
+
+def get_model_instances(model_name, app_label):
+    model = apps.get_model(app_label=app_label, model_name=model_name)
+    filter_args = {} if model_name == "PastEvent" else {"is_deleted": False}
+    model_instances = model.objects.filter(**filter_args)
+    return model_instances
+
+
+def auto_correct_spacing(instance, field_name, field_value):
+    for pattern in PATTERNS:
+        field_value = field_value.replace(pattern, "") 
+    setattr(instance, field_name, field_value)
+    instance.save()
+
+
+
+def is_feature_enabled(instance):
+    communities = db_models.Community.objects.filter(is_deleted=False)
+    flag = db_models.FeatureFlag.objects.filter(key=FEATURE_FLAG_KEY).first()
+    if not flag or not flag.enabled():
+        return False
+    enabled_communities = flag.enabled_communities(communities)
+    if hasattr(instance, "community") and instance.community in enabled_communities:
+         return True
+    elif hasattr(instance, "primary_community") and instance.primary_community in enabled_communities:
+         return True
+    else:
+         return False
```

Comment:
If the model has a "community" or "primary_community" field, but that is not filled, and the feature flag is enabled, it should make the change.  I can update the routine as follows:
`    if hasattr(instance, "community"):
        if not instance.community or instance.community in enabled_communities:
            return True
    elif hasattr(instance, "primary_community"):
        if not instance.primary_community or instance.primary_community in enabled_communities:
            return True
    return False
`